features/netsocket/NetworkInterface.h

```diff
@@ -147,7 +147,7 @@ class NetworkInterface: public DNS {
      *  @return  NSAPI_ERROR_IS_CONNECTED if interface is already connected.
      *  @return  negative error code on failure.
      */
-    virtual nsapi_error_t connect() = 0;
```

Comment:
I don't feel that this is the correct API change.

`NetworkInterface::connect()` has traditionally taken only configuration parameters and the non-parameter version is equivalent between all interfaces.

Now we are changing the behaviour of the non-parameter version, but not those that take parameters.


Can you point out the place in DNS resolver that utilises this knowledge of preferred IP stack version?